UpgradingACA-Py.md

```diff
@@ -0,0 +1,88 @@
+# Upgrading ACA-Py Data
+
+Some releases of ACA-Py may be improved by, or even require, an upgrade when
+moving to a new version. Such changes are documented in the [CHANGELOG.md],
+and those with ACA-Py deployments should take note of those upgrades. This
+document summarizes the upgrade system in ACA-Py.
+
+## Version Information and Automatic Upgrades
+
+The file [version.py] contains the current version of a running instance of
+ACA-Py. In addition, a record is made in the ACA-Py secure storage (database)
+about the "most recently upgraded" version. When deploying a new version of
+ACA-Py, the [version.py] value will be higher than the version in
+secure storage. When that happens, an upgrade is executed, and on successful
+completion, the version is updated in secure storage to match what is
+in [version.py].
+
+Upgrades are defined in the [Upgrade Definition YML file]. For a given
+version listed in the follow, the corresponding entry is what actions are
+required when upgrading from a previous version. If a version is not listed
+in the file, there is no upgrade defined for that version from its immediate
+predecessor version.
+
+Once an upgrade is identified as needed, the process is:
+
+- Collect (if any) the actions to be taken to get from the version recorded in
+secure storage to the current [version.py]
+- Execute the actions from oldest to newest.
+    - If the same action is collected more than once (e.g., "Resave the
+Connection Records" is defined for two different versions), perform the action
+only once.
+- Store the current ACA-Py version (from [version.py]) in the secure storage
+  database.
+
+## Forced Offline Upgrades
+
+In some cases, it may be necessary to do an offline upgrade, where ACA-Py is
+taken off line temporarily, the database upgraded explicitly, and then
+ACA-Py re-deployed as normal. As yet, we do not know of any cases of this, but
+those deploying ACA-Py should be aware of this possibility. For example,
+we may at some point need an upgrade that **MUST NOT** be executed by more
+than one ACA-Py instance. In that case, a "normal" upgrade could be dangerous
+for deployments on container orchestration platforms like Kubernetes.
+
+If the Maintainers of ACA-Py recognize a case where ACA-Py must be upgraded while
+offline, a new Upgrade feature will be added that will prevent the "auto upgrade"
+process from executing. See [Issue XXXX] for the status of that feature.
+
```

Comment:
I think we should also mention:
There are 2 flavours of this explicit upgrade mechanism, `"critical"` and `"warning"`. With `critical`, it will stop the "auto upgrade" process, raise an exception and force users to do the upgrade offline first. With `warning`, it will stop the "auto upgrade" process, log a warning but allows ACA-Py to startup normally. In this case, the version is not updated in secure storage and it will only update when doing the upgrade offline.

Reply:
I've referenced the docs. We'll update this once when the merge of the #2204 is done -- perhaps as part of that PR, even.